lib/open_feature/sdk/api.rb

```diff
@@ -42,10 +42,12 @@ def configure(&block)
         block.call(configuration)
       end
 
-      def build_client(name: nil, version: nil)
-        client_options = Metadata.new(name: name, version: version).freeze
-        provider = Provider::NoOpProvider.new if provider.nil?
-        Client.new(provider: provider, client_options: client_options, context: context)
+      def build_client(name: nil, version: nil, domain: nil)
+        client_options = Metadata.new(name: name, version: version, domain: domain).freeze
+
+        active_provider = provider(domain:).nil? ? Provider::NoOpProvider.new : provider(domain:)
```

Comment:
This is fine for now but eventually, the Ruby SDK should allow for providers to be bound to existing clients. That's how other SDKs behave but definitely adds to the complexity.